### PR TITLE
Feature/refactor android pay flow

### PR DESCRIPTION
**Pull request by sav007**
This PR simplifies the logic if AndroidPay enabled and ready on user device verification, all logic merged into one function `com.shopify.buy3.pay.PayHelper#isReadyToPay`.

As well it introduces new helper method `com.shopify.buy3.pay.PayHelper#handleWalletResponse` that simplifies handling onActivityResult for wallet requests:
```
PayHelper.handleWalletResponse(requestCode, resultCode, data, new PayHelper.WalletResponseHandler() {
      @Override public void onWalletError(final int requestCode, final int errorCode) {
        showError(-1, new RuntimeException("Failed wallet request, errorCode: " + errorCode));
      }

      @Override public void onMaskedWalletRequest() {
        requestMaskedWallet(googleApiClient);
      }

      @Override public void onMaskedWallet(final MaskedWallet maskedWallet) {
        updateMaskedWallet(maskedWallet);
      }

      @Override public void onFullWallet(final FullWallet fullWallet) {
        completeCheckout(fullWallet);
      }
    });
```

Refactors view presenters to not keep reference to google client.

### CODE REVIEW

File: MobileBuy/sample/src/main/java/com/shopify/sample/view/cart/CartCheckoutView.java [hunk 1]
```diff
@@ -84,21 +105,14 @@ public CartCheckoutView(@NonNull final Context context, @Nullable final Attribut
   }
 
   @Override public void showError(final long requestId, final Throwable t) {
-    //TODO show error message
+    //TODO log and show error message
```
Comment (dave-pelletier) [inline MobileBuy/sample/src/main/java/com/shopify/sample/view/cart/CartCheckoutView.java]: Is this a really old TODO comment? If we don't need this, might as well just delete the comment. If we do need it, can it go in this PR?

File: MobileBuy/sample/src/main/java/com/shopify/sample/view/cart/CartListView.java [hunk 2]
```diff
@@ -92,6 +92,8 @@ public CartListView(final Context context, final AttributeSet attrs) {
   }
 
   @Override public void showError(final long requestId, final Throwable t) {
+    //TODO log error
```
Comment (dave-pelletier) [inline MobileBuy/sample/src/main/java/com/shopify/sample/view/cart/CartListView.java]: another TODO

**Comment by dave-pelletier** [comment]
Lots of TODO comments

File: MobileBuy/sample/src/main/java/com/shopify/sample/view/cart/CartActivity.java [hunk 3]
```diff
@@ -63,9 +63,6 @@
 
   @Override protected void onActivityResult(final int requestCode, final int resultCode, final Intent data) {
     super.onActivityResult(requestCode, resultCode, data);
-
-    if (requestCode == PayHelper.REQUEST_CODE_MASKED_WALLET) {
```
Comment (sav007) [inline MobileBuy/sample/src/main/java/com/shopify/sample/view/cart/CartActivity.java]: Its handled in `PayHelper.handleWalletResponse`

**Comment by sav007** [comment]
TODO is related to sample that is not priority for now, and will be fixed next

**Pull request merged**